File: 1/1.c
#include <stdio.h>
#include <stdlib.h>

int main() {
    char *line = NULL;
    size_t size = 0;
    ssize_t len = 0;
    int sum = 0;

    FILE *fp = fopen("input", "rb");
    while((len = getline(&line, &size, fp)) != -1) {
        char *p = line;
        for (; !(*p >= '0' && *p <= '9'); p ++);

        sum += (*p - '0') * 10;

        p = line + len;
        for (; !(*p >= '0' && *p <= '9'); p --);

        sum += (*p - '0');

        free(line);
        size = 0;
    }
    fclose(fp);
    printf("%d\n", sum);
}

File: 1/1.py
with open("input", "r") as inp:
    sum = 0
    lines = inp.readlines()
    for line in lines:
        for c in line[::+1]:
            if c.isdigit():
                sum += (ord(c) - ord('0')) * 10
                break

        for c in line[::-1]:
            if c.isdigit():
                sum += ord(c) - ord('0')
                break

    print(sum)
